xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -153,6 +153,18 @@ impl DbConnection {
         Ok(groups.into_iter().next())
     }
 
+    /// Return a single group that matches the given welcome ID
+    pub fn find_group_by_welcome_id(
+        &self,
+        welcome_id: i64,
+    ) -> Result<Option<StoredGroup>, StorageError> {
+        let mut query = dsl::groups.order(dsl::created_at_ns.asc()).into_boxed();
+        query = query.limit(1).filter(dsl::welcome_id.eq(welcome_id));
```

Comment:
Can we log an error if there's more than one result?

Reply:
added here f7f0c746ce36e328c2e2e0ec7973677bac1cbb3b